### PR TITLE
Remove additional cast from TPCH q8

### LINKED ISSUE #6794: Incorrect results returned for TPC-H Query 8
### Describe the bug

Datafusion gives incorrect results when running TPC-H Query 8 with parquet files. 

### To Reproduce

1. Generate TPC-H parquet files for scale factor 10
2. Open datafusion-cli and create external tables pointing to files
```sql
create external table lineitem stored as parquet location '/path/to/lineitem/lineitem_1687987398_default/';
create external table customer stored as parquet location '/path/to/customer/customer_1687987384_default/';
create external table nation stored as parquet location '/path/to/nation/nation_1687988005_default/';
create external table orders stored as parquet location '/path/to/orders/orders_1687988005_default/';
create external table region stored as parquet location '/path/to/region/region_1687988223_default/';
create external table supplier stored as parquet location '/path/to/supplier/supplier_1687988223_default/';
create external table part stored as parquet location '/path/to/part/part_1687988133_default/';
```
3. Run TPC-H Query 8
```sql
select
  o_year, sum(case when nation = 'BRAZIL' then volume else 
0
 end) / sum(volume) as mkt_share
from
  (
    select
      extract(year from o_orderdate) as o_year,
      l_extendedprice * (
1
 - l_discount) as volume,
      n2.n_name as nation
    from part, supplier, lineitem, orders, customer, nation n1, nation n2, region
    where
      p_partkey = l_partkey
      and s_suppkey = l_suppkey
      and l_orderkey = o_orderkey
      and o_custkey = c_custkey
      and c_nationkey = n1.n_nationkey
      and n1.n_regionkey = r_regionkey
      and r_name = 'AMERICA'
      and s_nationkey = n2.n_nationkey
      and o_orderdate between date '1995-01-01' and date '1996-12-31'
      and p_type = 'ECONOMY ANODIZED STEEL'
  ) as all_nations
group by o_year
order by o_year;
```
4. Incorrect results displayed below
```
+--------+-------------------------------------------+
| o_year | mkt_share                                 |
+--------+-------------------------------------------+
| 1995.0 | -0.00000000000011380044067220119495060732 |
| 1996.0 | 0.00000000000019588288500717383285218261  |
+--------+-------------------------------------------+
2 rows in set. Query took 1.131 seconds.
```

### Expected behavior

The correct results should be:
1. From Postgres:
```
 o_year |       mkt_share        
--------+------------------------
   1995 | 0.03882014251433219622
   1996 | 0.03948968749183991638
(2 rows)

Time: 11925.416 ms (00:11.925)
```
2. From DuckDb (with same parquet files):
```
┌────────┬─────────────────────┐
│ o_year │      mkt_share      │
│ int64  │       double        │
├────────┼─────────────────────┤
│   1995 │  0.0388201425143322 │
│   1996 │ 0.03948968749183992 │
└────────┴─────────────────────┘
Run Time (s): real 1.328 user 8.095730 sys 0.358842
```

### Additional context

_No response_

### LINKED ISSUE #7216: TPC-H q8 fails with overflow (since version 28)
### Describe the bug

```
Arrow error: Compute error: Overflow happened on: 19293795273 * 100000000000000000000000000000000000000
```

### To Reproduce

Here is a Google colab notebook that demonstrates the issue

https://colab.research.google.com/drive/13vCffhO1-IP3NtXT1FgO-j6MwoQZlPtm?usp=sharing

### Expected behavior

_No response_

### Additional context

This was originally discovered by Mimoune Djouallah:

https://twitter.com/mim_djo/status/1686955130812542976?s=20

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #7216.
Closes #6794.

### CODE REVIEW

File: benchmarks/queries/q8.sql [hunk 1]
```diff
@@ -1,9 +1,9 @@
 select
     o_year,
-    cast(cast(sum(case
-                      when nation = 'BRAZIL' then volume
-                      else 0
-        end) as decimal(12,2)) / cast(sum(volume) as decimal(12,2)) as decimal(15,2)) as mkt_share
+    sum(case
+            when nation = 'BRAZIL' then volume
+            else 0
+        end) / sum(volume) as mkt_share
```
Comment (viirya) [inline benchmarks/queries/q8.sql:6]: These cast are added before to work around overflow issue.

After https://github.com/apache/arrow-datafusion/pull/6832, q8 should pass without cast.

Comment (viirya) [reply benchmarks/queries/q8.sql:6]: Looks okay, all tests are passed.

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @viirya

**Review by andygrove — APPROVED** [review]
Thanks @viirya

**Comment by viirya** [comment]
Thanks @alamb @andygrove

**Comment by alamb** [comment]
The code is flowing freely these days -- it is great to see!

**Pull request merged**